### DISCUSSION TITLE
Passkey auth

### DISCUSSION THREAD

**Discussion by pelmered** [filamentphp/filament#13623] [Ideas]
It would be great if we could have an option for passkey auth when logging in to Filament. 

Passkeys is the new way to authenticate securely and without all the hassle with passwords for both end users and service providers providing user support. While this could be done as a plugin, I really think this should be built in to Filament itself. This is most likely the future of auth, and Filament should of course stay on top of this. 

Here is a good guide on how this can be implemented: https://blog.joe.codes/implementing-passkey-authentication-in-your-laravel-app

**Comment by pelmered** [comment]
I just found a new Laracasts course about this as well: https://laracasts.com/series/add-passkeys-to-a-laravel-app

I don't think it would be that hard to implement yourself, but I still really thing this should be provided out of the box.

**Comment by altayevrim** [comment]
I second that. Now it is way easier to implement passkeys to the new package of spatie, https://github.com/spatie/laravel-passkeys but I think it should be officially added to filament.

Also I think some options to force passkey usage as well. (if a user has a passkey for example, an option to not allow password login)

**Reply by pelmered** [reply]
I think this will be added in the upcoming Filament 4. At least we will get MFA out of the box. This article about version 4 does not mention Passkeys specifically, but let's hope that we get support for that as well. 
https://filamentphp.com/content/alexandersix-all-about-the-filament-v4-beta-release#multi-factor-authentication

Maybe there wont be Passkey support from day one, but it sounds like a lot of the ground work will be in place for this at least.  Filament 4 will be introduced tomorrow at Laracon UK by the way!

**Reply by altayevrim** [reply]
Awesome news! And the funny thing is that I wasn't aware of the situation and the beta will be released tomorrow. I'm excited to test it out! Thanks for the news!
